src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/ApprovedPremisesSeedJob.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.seed
+
+import org.slf4j.LoggerFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.PropertyStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.PremisesRepository
+import java.util.UUID
+
+class ApprovedPremisesSeedJob(
+  fileName: String,
+  private val premisesRepository: PremisesRepository
+) : SeedJob<ApprovedPremisesSeedCsvRow>(
+  id = UUID.randomUUID(),
+  fileName = fileName,
+  requiredColumns = 12
+) {
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  override fun deserializeRow(columns: List<String>) = ApprovedPremisesSeedCsvRow(
+    id = UUID.fromString(columns[0]),
+    name = columns[1],
+    addressLine1 = columns[2],
+    postcode = columns[3],
+    totalBeds = Integer.parseInt(columns[4]),
+    notes = columns[4],
+    probationRegionId = UUID.fromString(columns[5]),
+    localAuthorityAreaId = UUID.fromString(columns[6]),
+    characteristicIds = columns[7].split(",").map { UUID.fromString(it.trim()) },
+    status = PropertyStatus.valueOf(columns[8]),
+    apCode = columns[9],
+    qCode = columns[10]
```

Comment:
Is it possible to address the columns using the column heading e.g. `columns.get("q_code")` rather than `columns[10]`? I have a bit of a horror of the ["connascence of position"](https://connascence.io/position.html) as I find it fragile to maintain and tiring mentally! I saw a [good talk on this years ago by Jim Weirich](https://www.youtube.com/watch?v=HQXVKHoUQxY)

Reply:
Hadn't heard of connascence before but I'll definitely be adding that one to the lexicon.

Interestingly, I had gone for indexes because I've had issues in the past with CSV uploads where the end user downloads the template (with headings correctly named) then decides to change the headers to something slightly different.

In this case though, since we'll be the ones uploading the CSV I'm with you on the index point - have adjusted accordingly.